third-party/sysdeps/common/gmp/CMakeLists.txt

```diff
@@ -0,0 +1,107 @@
+if(NOT CMAKE_SOURCE_DIR STREQUAL CMAKE_CURRENT_SOURCE_DIR)
+    # When included in TheRock, we download sources and set up the sub-project.
+    set(_source_dir "${CMAKE_CURRENT_BINARY_DIR}/source")
+    set(_download_stamp "${_source_dir}/download.stamp")
+
+    therock_subproject_fetch(therock-gmp-sources
+      SOURCE_DIR "${_source_dir}"
+      # Original project location: https://gmplib.org/
+      # Stable release used: 6.3.0
+      URL "https://rocm-third-party-deps.s3.us-east-2.amazonaws.com/gmp-6.3.0.tar.xz"
+      URL_HASH "SHA512=e85a0dab5195889948a3462189f0e0598d331d3457612e2d3350799dba2e244316d256f8161df5219538eb003e4b5343f989aaa00f96321559063ed8c8f29fd2"
+      TOUCH "${_download_stamp}"
+    )
+
+    therock_cmake_subproject_declare(therock-gmp
+      FPRINT_SOURCE_DIR "${_source_dir}"
+      FPRINT_FILE_GLOBS "${CMAKE_CURRENT_LIST_DIR}/*"
+      EXTERNAL_SOURCE_DIR .
+      BINARY_DIR build
+      NO_MERGE_COMPILE_COMMANDS
+      BACKGROUND_BUILD
+      OUTPUT_ON_FAILURE
+      CMAKE_ARGS
+        "-DSOURCE_DIR=${_source_dir}"
+        "-DPATCHELF=${PATCHELF}"
+        "-DPython3_EXECUTABLE=${Python3_EXECUTABLE}"
+      INSTALL_DESTINATION
+        lib/rocm_sysdeps
+      INTERFACE_LINK_DIRS
+        lib/rocm_sysdeps/lib
+      INTERFACE_INSTALL_RPATH_DIRS
+        lib/rocm_sysdeps/lib
+      INTERFACE_PKG_CONFIG_DIRS
+        lib/rocm_sysdeps/lib/pkgconfig
+      EXTRA_DEPENDS
+        "${_download_stamp}"
+    )
+    therock_cmake_subproject_provide_package(therock-gmp gmp lib/rocm_sysdeps/lib/cmake/gmp)
+    therock_cmake_subproject_activate(therock-gmp)
+
+    therock_test_validate_shared_lib(
+      PATH build/dist/lib/rocm_sysdeps/lib
+      LIB_NAMES libgmp.so
+    )
+
+    add_test(
+      NAME gmp_archive_test
+      COMMAND ls "${CMAKE_CURRENT_BINARY_DIR}/build/dist/lib/rocm_sysdeps/lib/libgmp.a"
+    )
+
+    add_test(
+      NAME gmp_header_test
+      COMMAND ls "${CMAKE_CURRENT_BINARY_DIR}/build/dist/lib/rocm_sysdeps/include/gmp.h"
+    )
+
+    return()
+endif()
+
+# Otherwise, this is the sub-project build.
+cmake_minimum_required(VERSION 3.25)
+project(GMP_BUILD VERSION "6.3.0")
+
+include(ProcessorCount)
+ProcessorCount(PAR_JOBS)
+
+if(CMAKE_SYSTEM_NAME STREQUAL "Linux" AND NOT PATCHELF)
+  message(FATAL_ERROR "Missing PATCHELF from super-project")
+endif()
+
+add_custom_target(
+  build ALL
+  WORKING_DIRECTORY "${CMAKE_CURRENT_BINARY_DIR}"
+  COMMAND
```

Comment:
Eventually worth to add
```cmake
  COMMAND
    "${CMAKE_COMMAND}" -E rm -rf -- "${CMAKE_INSTALL_PREFIX}" "${CMAKE_CURRENT_BINARY_DIR}/s"
```
as first command in the chain? Furthermore, looking at other sysdeps, we may want to copy the sources,